integration-tests/tests/src/tests/objects.ts

```diff
@@ -1335,4 +1335,38 @@ describe("Realm.Object", () => {
       expect(objFromKey?.age).equals(7);
     });
   });
+
+  // from https://github.com/realm/realm-js/issues/6129
+  describe("Primary key and property with default value", () => {
+    const PrimaryAndDefaultSchema: Realm.ObjectSchema = {
+      name: "MySchema",
+      primaryKey: "id",
+      properties: {
+        id: { type: "int" },
+        fieldOne: { type: "string" },
+        fieldTwo: { type: "string", default: "DEFAULT_VALUE" },
+      },
+    };
+    openRealmBeforeEach({ schema: [PrimaryAndDefaultSchema] });
+
+    [Realm.UpdateMode.All, Realm.UpdateMode.Modified].forEach((updateMode) => {
```

Comment:
I'm not a fan of the `forEach` being used like this, as it's undefined behaviour if the function argument is called synchronously or not by the runtime implementation and this code depends on it being called synchronously.

```suggestion
    for (const updateMode of [Realm.UpdateMode.All, Realm.UpdateMode.Modified]) {
```